src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UI/InfoBarService/VsInfoBarService.InfoBarEntry.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.VS.UI.InfoBarService;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore.InfoBarService;
+
+internal partial class VsInfoBarService
+{
+    private class InfoBarEntry : IVsInfoBarUIEvents
+    {
+        private readonly IVsInfoBarUIElement _element;
+        private readonly InfoBarUI[] _items;
+        private readonly Action<InfoBarEntry> _onClose;
+        private readonly uint cookie;
+
+        public InfoBarEntry(string message, IVsInfoBarUIElement element, InfoBarUI[] items, Action<InfoBarEntry> onClose)
+        {
+            Assumes.NotNull(message);
+            Assumes.NotNull(element);
+            Assumes.NotNull(items);
+            Assumes.NotNull(onClose);
+
+            Message = message;
```

Comment:
>Message [](http://example.com/codeflow?start=12&length=7)

normally, C# static analysis will throw a warning if fields are accessed without "this.". ISt his disabled in this codebase?

Reply:
note that I'm having a hrad time understanding this code without the this. prefixes. For instance, I assumed cookie was a local in this function.

Reply:
It looks like there is existing code that doesn't put in the "this." for property invocation, but fields all use the _ prefix.

Reply:
Fields should use the _ prefix.

Reply:
prefix added. Fixed in last commit